articles/web-scraping-in-python/index.md

```diff
@@ -0,0 +1,143 @@
+
+![Google forms class](/engineering-education/web-scraping-in-python/google-forms-class.png)
+
+As you can see in the image (highlighted portion), the class we are looking for is called *"freebirdFormviewerComponentsQuestionBaseTitle"*. If we want to obtain the descriptions of the questions (the text below the question), we can use "freebirdFormviewerComponentsQuestionBaseDescription" for the descriptions. However, In this article, we will only scrape the questions.
+
+```python
+def get_data(url):
+
+	# opening the url
+	url_client = urlopen(url)
+	# getting the content of the url.
+	url_content = url_client.read()
+	#close the connection
+	url_client.close()
+	# beautiful soup parser to parse the files in html.
+	parser = soup(url_content, "html.parser")
+	content = parser.findAll('div', {'class': 'freebirdFormviewerComponentsQuestionBaseTitle'})
+	# print(parser.name_of_tag) to print the first element of that particular tag.
+
+	return content
+```
+
+The get_data function takes the URL as a parameter and returns all the elements that belong to a particular class. The urlopen() function establishes a connection to the URL and the read() function reads the contents of that page. We create an object called parser using beautifulsoup. This function takes two parameters: the contents of the page and the type of parser. In this case, we shall use the HTML parser. the findall() method finds all the HTML elements based on a given condition. In this case, we are looking for all the "div" tags that have the class "freebirdFormviewerComponentsQuestionBaseTitle".
+
+#### Storing the data in an excel sheet
+Once the data gets fetched from the web page, the next step is to store it in a presentable format.
+
+```python
+def store_excel(data):
+
+	# Workbook() takes one, non-optional, argument  
+	# which is the filename that we want to create.
+	workbook = xlsxwriter.Workbook('content.xlsx')   
+	# The workbook object is then used to add new  
+	# worksheet via the add_worksheet() method.
+	worksheet = workbook.add_worksheet()   
+	# Use the worksheet object to write
+	# data via the write() method.
+	row = 0
+	col = 0
+	# parse the data and store it in the excel sheet.
+	for item in data:
+
+		text = str(item.text)
+		worksheet.write(row, col, text)
+		row += 1
+	# Finally, close the Excel file
+	# via the close() method.
+	workbook.close()
+```
+
+The store_excel() function takes the data as a parameter and creates an excel sheet that contains the data. The Workbook() function creates a workbook with the name "content.xlsx" and add_worksheet() adds a new worksheet to our workbook. We then go through all the items in our data (all the questions) and add them to our worksheet using the write() method. The row and column integers denote the current row and column in the sheet.
+
+#### Putting it all together
+
+```python
+if __name__ == '__main__':
+
+	url = input('enter the url: ')
+	data = get_data(url)
+	print(data)
```

Comment:
Possibly redundant output of the div

Reply:
Hey @glennslaven , I have made the changes you requested. Do you mean the print statement is redundant here? I have removed it.